src/economy/portdock.cc

```diff
@@ -502,6 +382,32 @@ uint32_t PortDock::count_waiting(const PortDock* dest) const {
 	return waiting_.size();
 }
 
+uint32_t PortDock::calc_max_priority(const EditorGameBase& egbase, const PortDock& dest) const {
+	uint32_t p = 0;
+	for (const ShippingItem& si : waiting_) {
+		if (si.destination_dock_.serial() == dest.serial()) {
```

Comment:
Why cant we comapre the PortDocks directly?

Reply:
https://www.widelands.org/documentation/development/#why-std-set-somepointer-is-a-bad-idea